cve_bin_tool/output_engine/__init__.py

```diff
@@ -0,0 +1,122 @@
+import csv
+import json
+import os
+
+from datetime import datetime
+from logging import Logger
+from typing import List, Dict, Set
+
+from ..cve_scanner import CVEData
+from ..log import LOGGER
+from .console import output_console
+from .html import output_html
+from .util import generate_filename
+from .util import format_output
+from .util import add_extension_if_not
+
+
+def output_json(all_cve_data, outfile):
+    """ Output a JSON of CVEs """
+    formatted_output = format_output(all_cve_data)
+    json.dump(formatted_output, outfile, indent="    ")
+
+
+def output_csv(all_cve_data, outfile):
+    """ Output a CSV of CVEs """
+    formatted_output = format_output(all_cve_data)
+    writer = csv.DictWriter(
+        outfile,
+        fieldnames=[
+            "vendor",
+            "product",
+            "version",
+            "cve_number",
+            "severity",
+            "remarks",
+        ],
+    )
+    writer.writeheader()
+    writer.writerows(formatted_output)
+
+
+class OutputEngine:
+    def __init__(
+        self,
+        all_cve_data: Set[CVEData],
+        filename: str,
+        logger: Logger = None,
+        products_with_cve: int = 0,
+        products_without_cve: int = 0,
+        total_files: int = 0,
+    ):
+        self.logger = logger or LOGGER.getChild(self.__class__.__name__)
+        self.all_cve_data = all_cve_data
+        self.filename = os.path.abspath(filename) if filename else ""
+        self.products_with_cve = products_with_cve
+        self.products_without_cve = products_without_cve
+        self.total_files = total_files
+
+    def output_cves(self, outfile, output_type=None):
```

Comment:
It would be good to initialize output_type with "console" for better readability